File: tests/fixtures/repos/java-shop/src/main/java/com/acme/orders/OrderService.java
package com.acme.orders;

import org.springframework.stereotype.Service;

@Service
public class OrderService {
    private final OrderRepository repository;
    private final PaymentGateway payments;

    public OrderService(OrderRepository repository, PaymentGateway payments) {
        this.repository = repository;
        this.payments = payments;
    }

    public Order create(CreateOrderRequest request) {
        Order order = new Order();
        payments.charge(order.getTotal());
        return repository.save(order);
    }

    public Order find(Long id) {
        return repository.findById(id).orElse(null);
    }
}

File: tests/fixtures/repos/ts-shop/src/orders/order.service.ts
import { Injectable } from '@nestjs/common';
import { OrderRepository } from './order.repository';
import { PaymentService } from '../payments/payment.service';
import { InventoryService } from '../inventory/inventory.service';
import { Logger } from '../shared/logger';
import { CreateOrderDto } from './dto';
import { Order } from './order.entity';

@Injectable()
export class OrderService {
  constructor(
    private readonly repo: OrderRepository,
    private readonly payments: PaymentService,
    private readonly inventory: InventoryService,
    private readonly logger: Logger,
  ) {}

  async createOrder(dto: CreateOrderDto): Promise<Order> {
    this.logger.info('creating order');
    const order = new Order();
    order.total = dto.total;
    this.validate(order);
    await this.inventory.reserve(dto.items);
    await this.payments.charge(order);
    order.status = 'confirmed';
    return this.repo.save(order);
  }

  async findOrder(id: string): Promise<Order | null> {
    return this.repo.findById(id);
  }

  private validate(order: Order): void {
    if (order.total < 0) throw new Error('invalid');
  }
}
